test/gen-server/ApiServer.ts

```diff
@@ -1274,6 +1274,21 @@ describe('ApiServer', function() {
     assert.deepEqual(resp.data?.options, undefined);
   });
 
+  it('PATCH /api/docs/{did} supports proper values for type key', async function() {
+    const did = await dbManager.testGetId('Surprise2');
+    // Check setting null for normal document type
+    let resp = await axios.patch(`${homeUrl}/api/docs/${did}`, {type: DOCTYPE_NORMAL}, chimpy);
+    assert.equal(resp.status, 200);
+
+    // check setting template as document type
+    resp = await axios.patch(`${homeUrl}/api/docs/${did}`, {type: DOCTYPE_TEMPLATE}, chimpy);
+    assert.equal(resp.status, 200);
+
+    // check setting tutorial as document type
+    resp = await axios.patch(`${homeUrl}/api/docs/${did}`, {type: DOCTYPE_TUTORIAL}, chimpy);
+    assert.equal(resp.status, 200);
```

Comment:
I think we should also check the save went through by calling `axios.get(...)` after.

Also, since these 3 assertions all follow the same pattern of passing a `type` and asserting the `type` is present, we can refactor this to use a for loop.